### ISSUE TITLE
[Help Wanted] [PLT-5667] Automatically convert usernames to lowercase letters on sign in

### ISSUE THREAD

**Issue by esethna** [mattermost/platform#5585]
If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://pre-release.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://docs.mattermost.com/guides/developer.html), specifically for [machine setup](https://docs.mattermost.com/developer/developer-setup.html) and for [developer workflow](https://docs.mattermost.com/developer/developer-flow.html).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/PLT-5667)

Automatically convert usernames to lowercase letters on sign in

Repro of issue:
1) Create an account with username "Jason"
  -- username is converted to lowercase ("jason") in the database
2) Try to log in with that username 

Observed: "We couldn't match your credentials" error
Expected: Username converted to lowercase ("jason") on sign in so login is successful

**Comment by cpanato** [comment]
I can work on that

**Comment by cpanato** [comment]
@esethna I created a user with username Carlos, and after that, I was able to login using Carlos or carlos as a username in the login page. Both works.

I used master as base.

**Comment by jasonblais** [comment]
@cpanato I tried on `master` and I get the following 

![image](https://cloud.githubusercontent.com/assets/13119842/23512996/9af9f0da-ff30-11e6-8ecf-3a6aa3bdb3e8.png)

But signing in with `jason123456` works.

**Comment by cpanato** [comment]
@jasonblais let me try again

**Comment by cpanato** [comment]
@jasonblais 
![lmvjpbrj1v](https://cloud.githubusercontent.com/assets/4115580/23513604/d915539e-ff64-11e6-8e20-cadc6d62b39b.gif)

**Comment by jasonblais** [comment]
Hmm, interesting. It looks like it only reproduces on a PostgreSQL server.

If I try on MySQL, it works fine. Is your local environment on MySQL?

**Comment by cpanato** [comment]
I have both docker postgres/mysql.
now the question :) : how can I use mysql instead of postgress ? @jasonblais 
there are any docs about that?

**Comment by cpanato** [comment]
@jasonblais is it in the config.json?

```
"SqlSettings": {
        "DriverName": "mysql",
        "DataSource": "mmuser:mostest@tcp(dockerhost:3306)/mattermost_test?charset=utf8mb4,utf8",
        "DataSourceReplicas": [],
        "MaxIdleConns": 20,
        "MaxOpenConns": 300,
        "Trace": false,
        "AtRestEncryptKey": "sgrkr8r3xr5k4ux57t6y5minc79fedgf"
    },
```

what is the config for postgres ?

**Comment by jasonblais** [comment]
We don't currently have support for migrating between databases automatically, as far as I know.

But I would ask in the [developers channel on pre-release](https://pre-release.mattermost.com/core/channels/developers), as they might know.

**Comment by cpanato** [comment]
this issue happen when using Postgres. 
I will fix that on the webapp to make lowercase the loginid

**Comment by cpanato** [comment]
PR - https://github.com/mattermost/platform/pull/5613

**Comment by realrolfje** [comment]
This problem seems re-introduced in 3.7.2 running on PostgreSQL.

**Comment by jasonblais** [comment]
Thanks @realrolfje for the report! The PR will actually be included in Mattermost 3.8 to be released on April 16th, so it's coming soon :)

https://github.com/mattermost/platform/pull/5613
